src/graphapi/pointwise.cpp

```diff
@@ -0,0 +1,251 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/errors.hpp>
+#include <miopen/graphapi/pointwise.hpp>
+
+namespace miopen {
+
+namespace graphapi {
+
+Pointwise PointwiseBuilder::build()
+{
+    if(!mModeSet || !mMathPrecisionSet)
+    {
+        MIOPEN_THROW(miopenStatusBadParm);
+    }
+    return mPointwise;
+}
+
+void BackendPointwiseDescriptor::setAttribute(miopenBackendAttributeName_t attributeName,
+                                              miopenBackendAttributeType_t attributeType,
+                                              int64_t elementCount,
+                                              void* arrayOfElements)
+{
+    if(mFinalized)
+    {
+        MIOPEN_THROW(miopenStatusNotInitialized);
```

Comment:
I am not sure `miopenStatusNotInitialized` is a good error code here. `miopenStatusInvalidValue` is probably a proper one here.

Reply:
Agree with you, but I decided to return the same error as described in [CUDNN Graph API docs](https://docs.nvidia.com/deeplearning/cudnn/api/cudnn-graph-library.html#cudnnbackendsetattribute). Do you think we should deviate from it here?

Reply:
I suggest we should not until unless we have a reason for that.

Reply:
It is missleading, but I guess it should be OK